extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamExtractor.java

```diff
@@ -178,6 +178,39 @@ public StreamExtractor(StreamingService service, LinkHandler linkHandler) {
     @Nonnull
     public abstract String getUploaderAvatarUrl() throws ParsingException;
 
+    /**
+     * The Url to the page of the parent chanel of the stream. This must not be a homepage,
+     * but the page offered by the service the extractor handles. This url will be handled by the
+     * <a href="https://teamnewpipe.github.io/documentation/03_Implement_a_service/#channel">ChannelExtractor</a>,
+     * so be sure to implement that one before you return a value here, otherwise NewPipe will crash if one selects
+     * this url.
+     *
+     * @return the url to the page of the parent chanel of the stream or an empty String
+     * @throws ParsingException
+     */
+    @Nonnull
+    public abstract String getParentChannelUrl() throws ParsingException;
```

Comment:
Should we add a `getParentChannelUrl` or a `getSubchannelUrl` here? Same thing for name and avatar URL obviously,

Imho we should add `getSubchannelUrl`, as currently we return the parent channel's URL, so the values returned by `getChannelUrl` stays the same. I want someone else's opinion here first though.

Edit: it looks like you're returning the subchannel's URL here, so that's obviously not correct anyway.

Reply:
I'm not really sure myself what will be the right naming. In Peertube it's "account page" and "channel page", maybe we should do the same?
![image](https://user-images.githubusercontent.com/21021560/79267134-4c1c9f00-7ea1-11ea-8aee-2139d3fe3c44.png)

Reply:
@Royosef: No, just change it to `getSubchannelUrl()`. We aren't going to call it accounts and channels as that's unclear and only PeerTube calls it that.